### ISSUE TITLE
"Disappearing Messages" feature

### ISSUE THREAD

**Issue by dbr** [boxdot/gurk-rs#79]
Currently if you have Disappearing Messages enabled in a chat, they are disabled as soon as a message is sent from gurk to that chat.

I guess this is because the client doesn't support the Disappearing Messages feature, rather than it specifically disabling the option

**Comment by boxdot** [comment]
Oh, yes. We don't handle any special of groups yet. We should definitely look into that.

**Comment by freswa** [comment]
It's probably worth to wait for upstream to implement this feature, so that presage gains some better handling maybe: https://gitlab.com/whisperfish/whisperfish/-/issues/166

**Comment by nothingbutlucas** [comment]
Is there any way to delete messages manually? At least to be able to make a workaround with a script that deletes the messages every x time or something like that?

**Comment by nothingbutlucas** [comment]
Well, I found this way to do [something](https://github.com/nothingbutlucas/clean-gurk) that works for me for the moment (script + cron). It would still be nice to have this functionality :)

**Comment by freswa** [comment]
If anyone has the time to work on this, I'd love to sponsor the feature.

**Comment by cwedgwood** [comment]
> If anyone has the time to work on this, I'd love to sponsor the feature.

@freswa I have a working implementation of this in PR #522 (depends on #519 for message deletion support).

Testing and feedback welcome - the code is functional but may have edge cases.

**Database note:** This adds new columns to `gurk.sqlite`. Back up `~/.local/share/gurk/gurk.sqlite` before upgrading - to revert, stop gurk and restore the backup. If you prefer a clean start, deleting `gurk.sqlite` (NOT `signal.sqlite`) will rebuild it from scratch. You stay linked, just lose local message history in gurk.

Build from source: `cargo install --git https://github.com/cwedgwood/gurk-rs --branch cw/all-features gurk`

Pre-built binaries (Linux, macOS) are also available from the CI artifacts: https://github.com/cwedgwood/gurk-rs/actions/runs/23992955668
